relay-sampling/src/evaluation.rs

```diff
@@ -195,19 +195,21 @@ impl<'a> SamplingEvaluator<'a> {
         I: Iterator<Item = &'b SamplingRule>,
     {
         for rule in rules {
-            if !rule.condition.matches(instance) {
+            if !rule.time_range.contains(self.now) || !rule.condition.matches(instance) {
                 continue;
             };
 
-            let Some(sampling_value) = rule.evaluate(self.now, self.reservoir) else {
+            let Some(sampling_value) = rule.adjusted_sampling_value(self.now) else {
                 continue;
             };
 
-            self.rule_ids.push(rule.id);
```

Comment:
I had to move this into the arms of `Factor` and `SampleRate` because previously if a reservoir rule didn't match the `rule.evaluate` would have returned None and continued in the lines above. 

If I left it here, the reservoir rules would have been part of the matched_ids for a `SampleRate` match